integration_tests/route_helpers.go

```diff
@@ -97,3 +105,36 @@ func clearRoutes() {
 	routerDB.C("routes").DropCollection()
 	routerDB.C("backends").DropCollection()
 }
+
+func getMongoSession(uri string) *mgo.Session {
```

Comment:
Agreed. It might also be nice to separate out the creation of the `mgo.DialInfo` value, and the creation of the connection _using_ that value. Doing that will also help with testability: you can test the various permutations of things that will manipulate the `DialInfo` value independently of testing that creation of a connection with _any_ `DialInfo` value.